src/quickdocumentdialog.cpp

```diff
@@ -491,46 +520,81 @@ void QuickDocumentDialog::addUserPaper()
 	}
 }
 
-void QuickDocumentDialog::addUserInputEncoding()
+void QuickDocumentDialog::addUserFontEncoding()
 {
 	QString newoption;
 	UniversalInputDialog dialog;
 	dialog.addVariable(&newoption, tr("New:"));
 	if (dialog.exec() && !newoption.isEmpty()) {
-		otherInputEncodingList.append(newoption);
+		otherFontEncodingList.append(newoption);
 		Init();
 	}
 }
 
-void QuickDocumentDialog::addUserFontEncoding()
+void QuickDocumentDialog::addBabelOption()
 {
 	QString newoption;
 	UniversalInputDialog dialog;
 	dialog.addVariable(&newoption, tr("New:"));
 	if (dialog.exec() && !newoption.isEmpty()) {
-		otherFontEncodingList.append(newoption);
+		otherBabelOptionsList.append(newoption);
 		Init();
 	}
 }
 
-void QuickDocumentDialog::addBabelOption()
+void QuickDocumentDialog::addUserOptions()
 {
 	QString newoption;
 	UniversalInputDialog dialog;
 	dialog.addVariable(&newoption, tr("New:"));
 	if (dialog.exec() && !newoption.isEmpty()) {
-		otherBabelOptionsList.append(newoption);
+		otherOptionsList.append(newoption);
 		Init();
 	}
 }
 
-void QuickDocumentDialog::addUserOptions()
+void QuickDocumentDialog::addUserPackages()
 {
 	QString newoption;
 	UniversalInputDialog dialog;
 	dialog.addVariable(&newoption, tr("New:"));
 	if (dialog.exec() && !newoption.isEmpty()) {
-		otherOptionsList.append(newoption);
+		otherPackagesList.append(newoption);
 		Init();
 	}
 }
+//when package selections change
+void QuickDocumentDialog::updateCheckState()
```

Comment:
I am not sure what exactly the logic is supposed to do, but basically if you select a row by clicking on a description text, all other selections/checked states are removed.
Let's stick with Qt logic, i.e. you click on the box to select something to be inserted ...

Reply:
You may try what happens if you remove my connected signals (I started development without these). But I found that checking more than three items is cumbersome since the checkboxes have a small size and it's slow (remember the project offering all known packages). Selecting by dragging the pointer over the rows or clicking into the (large) rows is quick and easy. Since this is a table (not a list) you have to use modifiers as it is known from windows file explorer or calc/excel. Thus clicking one row results in a selection containing only this row. Right, if you slightly miss a checkbox you want to click on then this row will be the only one selected. But you can redo your selection quickly and take care the next time. Maybe we could add the tooltip text before the table so help is immediately visible (s. image)?

![grafik](https://user-images.githubusercontent.com/102688820/212207279-2aaf5cad-0f5b-4ff4-a9ac-baf86aec3ca3.png)

If the wizard starts with initial selections (maybe a longer table with a scrollbar) the selected lines make it easier to recognize which rows are selected than the checkboxes on their own would do. And you can go easily through the descriptions of the selected packages without moving your eyes from left to right.

I also investigated the case when I keep the selection theme but remove the checkboxes at all. But at the first clicks I was not convinced that this is enough feedback for everyone to know that one already selected packages. So I added the checkboxes again which give a pretty clear feedback. But I don't use them for selecting for the reasons said, but you can if you want to.

This image may clarify why I wanted to synchronize row selections and checkStates:

![grafik](https://user-images.githubusercontent.com/102688820/212198900-42c27462-0e17-4077-8d06-4cbd2641c463.png)

Compare to this (I just expanded the previous selection, no signals):

![grafik](https://user-images.githubusercontent.com/102688820/212200321-cfb73d20-2e32-40f0-9a49-2d673bc970c6.png)

So I would recommend allowing selections.

Reply:
You may try what happens if you remove my connected signals (I started development without these). But I found that checking more than three items is cumbersome since the checkboxes have a small size and it's slow (remember the project offering all known packages). Selecting by dragging the pointer over the rows or clicking into the (large) rows is quick and easy. Since this is a table (not a list) you have to use modifiers as it is known from windows file explorer or calc/excel. Thus clicking one row results in a selection containing only this row. Right, if you slightly miss a checkbox you want to click on then this row will be the only one selected. But you can redo your selection quickly and take care the next time. Maybe we could add the tooltip text before the table so help is immediately visible (s. image)?

![grafik](https://user-images.githubusercontent.com/102688820/212207279-2aaf5cad-0f5b-4ff4-a9ac-baf86aec3ca3.png)

If the wizard starts with initial selections (maybe a longer table with a scrollbar) the selected lines make it easier to recognize which rows are selected than the checkboxes on their own would do. And you can go easily through the descriptions of the selected packages without moving your eyes from left to right.

I also investigated the case when I keep the selection theme but remove the checkboxes at all. But at the first clicks I was not convinced that this is enough feedback for everyone to know that one already selected packages. So I added the checkboxes again which give a pretty clear feedback. But I don't use them for selecting for the reasons said, but you can if you want to.

This image may clarify why I wanted to synchronize row selections and checkStates:

![grafik](https://user-images.githubusercontent.com/102688820/212198900-42c27462-0e17-4077-8d06-4cbd2641c463.png)

Compare to this (I just expanded the previous selection, no signals):

![grafik](https://user-images.githubusercontent.com/102688820/212200321-cfb73d20-2e32-40f0-9a49-2d673bc970c6.png)

So I would recommend allowing selections.